libs/@local/hash-graph-sdk/python/graph_sdk/types.py

```diff
@@ -0,0 +1,207 @@
+"""Create dynamic types from entity, property, and data types via pydantic."""
+from uuid import UUID
+
+from graph_types import (
+    DataTypeSchema,
+    EntityTypeSchema,
+    PropertyTypeSchema,
+)
+from yarl import URL
+
+from graph_sdk.client.concurrent import HASHClient, with_actor
+from graph_sdk.filter import DataTypePath, EntityTypePath, PropertyTypePath
+from graph_sdk.options import Options
+from graph_sdk.query import Parameter
+from graph_sdk.utils import (
+    async_to_sync,
+    filter_latest_ontology_types_from_subgraph,
+)
+
+
+class TypeAPI:
+    """GraphAPI for use with hash-graph-types."""
+
+    inner: HASHClient
+
+    def __init__(self, base: URL) -> None:
+        """Initialize the client with the base URL."""
+        self.inner = HASHClient(base)
+
+    async def load_data_type(
+        self,
+        data_type_id: str,
+        *,
+        actor_id: UUID,
+    ) -> DataTypeSchema:
+        """Load an external data type."""
+        with with_actor(self.inner, actor_id):
+            await self.inner.load_external_data_type(
+                URL(data_type_id),
+            )
+
+        return await self.get_data_type(
+            data_type_id,
+            actor_id=actor_id,
+            is_after_load=True,
+        )
+
+    async def get_data_type(
+        self,
+        data_type_id: str,
+        *,
+        actor_id: UUID,
+        is_after_load: bool = False,
+    ) -> DataTypeSchema:
+        """Returns the data type schema for the given data type ID.
+
+        If the data type is not found it will attempt to fetch it and use
+        the actor ID to authenticate the request.
```

Comment:
This is supposed to be implemented in the Graph API. Could you add a TODO to remove this once [H-136](https://linear.app/hash/issue/H-136) is done? (Same for property types and entity types)

Reply:
addressed in [577146f](https://github.com/hashintel/hash/pull/2868/commits/577146f5ba49ea34e9db3af7ae67a7fc9a47c267)